builder/googlecompute/builder.go

```diff
@@ -48,28 +48,58 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 	state.Put("hook", hook)
 	state.Put("ui", ui)
 
-	// Build the steps.
-	steps := []multistep.Step{
-		new(StepCheckExistingImage),
-		&StepCreateSSHKey{
-			Debug:        b.config.PackerDebug,
-			DebugKeyPath: fmt.Sprintf("gce_%s.pem", b.config.PackerBuildName),
-		},
-		&StepCreateInstance{
-			Debug: b.config.PackerDebug,
-		},
-		&StepInstanceInfo{
-			Debug: b.config.PackerDebug,
-		},
-		&communicator.StepConnect{
-			Config:    &b.config.Comm,
-			Host:      commHost,
-			SSHConfig: sshConfig,
-		},
-		new(common.StepProvision),
-		new(StepWaitInstanceStartup),
-		new(StepTeardownInstance),
-		new(StepCreateImage),
+	var steps []multistep.Step
+
+	if b.config.Comm.Type == "winrm" {
+
+		// Build the steps.
+		steps = []multistep.Step{
+			new(StepCheckExistingImage),
+			&StepCreateInstance{
+				Debug: b.config.PackerDebug,
+			},
+			&StepCreateWindowsPassword{
```

Comment:
What is the preferred method for steps that would only run on certain instance types?

In the step itself I could use the Image.IsWindows() and just do nothing instances where it doesn't apply?